validator/project_validator.go

```diff
@@ -655,24 +664,23 @@ func validateBVFields(project *model.Project) ValidationErrors {
 
 		hasTaskWithoutDistro := false
 		for _, task := range buildVariant.Tasks {
-			taskHasValidDistro := false
-			for _, d := range task.RunOn {
-				if d != "" {
-					taskHasValidDistro = true
-					break
-				}
-			}
-			if !taskHasValidDistro {
-				// check for a default in the task definition
-				pt := project.FindProjectTask(task.Name)
-				if pt != nil {
-					for _, d := range pt.RunOn {
-						if d != "" {
+			taskHasValidDistro := validateRunOn(task.RunOn)
+			if !taskHasValidDistro && task.IsGroup {
+				for _, t := range project.FindTaskGroup(task.Name).Tasks {
+					pt := project.FindProjectTask(t)
+					if pt != nil {
+						if validateRunOn(pt.RunOn) {
 							taskHasValidDistro = true
 							break
 						}
 					}
 				}
+			} else if !taskHasValidDistro {
+				// check for a default in the task definition
+				pt := project.FindProjectTask(task.Name)
+				if pt != nil {
+					taskHasValidDistro = validateRunOn(pt.RunOn)
+				}
 			}
 			if !taskHasValidDistro {
 				hasTaskWithoutDistro = true
```

Comment:
Sorry I'm just asking you to refactor this now but we don't need hasTaskWithoutDistro either, since we just break when we find it; we can just move the error at 691 into this and _then_ break

Reply:
all good and tests pass